tests/test_job_log_cursor.py

```
@@ -0,0 +1,113 @@
+forever once N reached the deque maxlen — causing the log stream to silently
+cut out after ~1000 lines (~5-7 min on a 4090).
+"""
+import pytest
+from services.jobs.job import Job
+from services.models.job import JobType
+
+
+def make_job(n_lines: int, maxlen: int = 2000) -> Job:
+    """Create a job and write n_lines log entries into it."""
+    j = Job(job_id="test", job_type=JobType.TRAINING)
+    j.logs.maxlen  # confirm deque is initialised
```

Ruff (0.15.13)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)